### PR TITLE
Add test cases from the case study regarding transitive change propagation

### PR DESCRIPTION
**Pull request by tsaglam**
This PR adds the test project that tests the transitive change propagation with a network of transformations (change propagation specification) between PCM, UML Class, and Java. The test cases are based on the PCM and UML Class test cases but they are extended to check whether the Java code is consistent as well. As the case study did not fix all the discovered issues, some test cases are executed with the linear network, meaning PCM ↔︎ UML ↔︎ Java.

These are the test cases using the linear network right now:
- AttributeConceptTest
- SignatureConceptTest
- ParameterConceptTest
- RequiredRoleConceptTest

These are the test cases using the fully connected network:
- AssemblyContextConceptTest
- CompositeDataTypeConceptTest
- InterfaceConceptTest
- ProvidedRoleTest
- RepositoryComponentConceptTest
- RepositoryConceptTest
- SystemConceptTest

### CODE REVIEW

File: tests/tools.vitruv.applications.pcmumlclass.tests/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -4,11 +4,15 @@ Bundle-Name: Vitruv PCM-UML Class Application Tests
 Bundle-SymbolicName: tools.vitruv.applications.pcmumlclass.tests
 Bundle-Version: 0.2.0.qualifier
 Automatic-Module-Name: tools.vitruv.applications.pcmumlclass.tests
-Fragment-Host: tools.vitruv.applications.pcmumlclass
```
Comment (tsaglam) [inline tests/tools.vitruv.applications.pcmumlclass.tests/META-INF/MANIFEST.MF:7]: `pcmumlclass.tests` is no longer a fragment to allow proper dependencies on it.

File: tests/tools.vitruv.applications.transitivechange.tests/src/tools/vitruv/applications/pcmumlclassjava/LinearTransitiveChangeTest.xtend [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * used applications contain stil some issues regarding the fully transitive use case.
+ * TODO TS: these issues should be fixed in the PCM/UML/Java applications and all sub classes should extend TransitiveChangeTest directly
+ */
+abstract class LinearTransitiveChangeTest extends TransitiveChangeTest {
+
+    @BeforeAll
+    def static void enforceLinearNetwork() {
+        linearNetwork = true
+    }
+}
```
Comment (tsaglam) [inline tests/tools.vitruv.applications.transitivechange.tests/src/tools/vitruv/applications/pcmumlclassjava/LinearTransitiveChangeTest.xtend:17]: This is the workaround which allows that some test to only use the linear network, while the test whose issues were completely fixed still use the fully connected network.

File: tests/tools.vitruv.applications.umljava.tests/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -5,9 +5,16 @@ Bundle-SymbolicName: tools.vitruv.applications.umljava.tests
 Automatic-Module-Name: tools.vitruv.applications.umljava.tests
 Bundle-Version: 0.2.0.qualifier
 Bundle-RequiredExecutionEnvironment: JavaSE-11
-Fragment-Host: tools.vitruv.applications.umljava
```
Comment (tsaglam) [inline tests/tools.vitruv.applications.umljava.tests/META-INF/MANIFEST.MF:8]: `umljava.tests` is also no longer a fragment to allow proper dependencies on it.

File: tests/tools.vitruv.applications.pcmumlclass.tests/META-INF/MANIFEST.MF (hunk 1 above)
Comment (HeikoKlare) [reply tests/tools.vitruv.applications.pcmumlclass.tests/META-INF/MANIFEST.MF:7]: This is fine for now and should be reverted as soon as we finally cleanup all the uilities (see #3).

File: tests/tools.vitruv.applications.umljava.tests/META-INF/MANIFEST.MF (hunk 3 above)
Comment (HeikoKlare) [reply tests/tools.vitruv.applications.umljava.tests/META-INF/MANIFEST.MF:8]: This is fine for now and should be reverted as soon as we finally cleanup all the uilities (see #3).

File: bundles/tools.vitruv.applications.umljava/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -5,24 +5,15 @@ Bundle-SymbolicName: tools.vitruv.applications.umljava;singleton:=true
...
  edu.kit.ipd.sdq.activextendannotations,
  tools.vitruv.domains.uml,
  tools.vitruv.domains.java,
  tools.vitruv.framework.applications,
  tools.vitruv.framework.change,
- tools.vitruv.framework.change.processing,
- tools.vitruv.framework.correspondence,
- tools.vitruv.framework.userinteraction,
- tools.vitruv.framework.util,
- tools.vitruv.extensions.dslsruntime.reactions,
```
Comment (HeikoKlare) [inline bundles/tools.vitruv.applications.umljava/META-INF/MANIFEST.MF:24]: These dependencies are required by the Reactions. Removing them will throw warnings.

**Review by HeikoKlare — CHANGES_REQUESTED** [review]

File: bundles/tools.vitruv.applications.umljava/META-INF/MANIFEST.MF (hunk 4 above)
Comment (HeikoKlare) [reply bundles/tools.vitruv.applications.umljava/META-INF/MANIFEST.MF:24]: This also applied to `com.google.guava`.

**Review by HeikoKlare — APPROVED** [review]

**Pull request merged**